compiler/src/iree/compiler/Dialect/HAL/Transforms/PruneExecutables.cpp

```diff
@@ -59,12 +60,32 @@ static void processOp(Operation *op, SymbolReferenceMap &referenceMap) {
   }
 }
 
+// Returns true if the users treating a symbol as a fallback are transitively
+// reachable directly or as fallbacks for reachable symbols.
+static bool reachableAsFallback(ArrayRef<Attribute> users,
+                                SymbolReferenceMap &referenceMap) {
+  for (auto user : users) {
+    if (referenceMap[user].count > 0) {
+      // The user treating the given symbol as a fallback is used directly.
+      return true;
+    }
+    if (!reachableAsFallback(referenceMap[user].fallbackOf, referenceMap)) {
+      // The user treating the given symbol as a fallback itself is a fallback
+      // that has uses.
+      return false;
```

Comment:
```suggestion
    if (reachableAsFallback(referenceMap[user].fallbackOf, referenceMap)) {
      // The user treating the given symbol as a fallback itself is a fallback
      // that has uses.
      return true;
```
Getting unexpected pruning here. Changing to this matches what the comment says, also returning false in the loop seems wrong no matter what because we could skip a valid user, even if this one is not.

Reply:
totally correct!